app/models/vm_or_template.rb

```diff
@@ -716,6 +716,27 @@ def disconnect_storage(s = nil)
     end
   end
 
+  # Return a list of parent compute resources (Resource Pool/Host/Cluster)
+  def parent_resources(options = {})
+    with_relationship_type('ems_metadata') do
+      path = []
+
+      # Can't use ancestors on VM because of multiple parents, so start with the VM's
+      # resource pool (first component of the host&clusters view)
+      rp = parent(:of_type => "ResourcePool")
+      unless rp.nil?
+        path = rp.ancestors + [rp]
+        path.reject! { |p| p.respond_to?(:hidden?) && p.hidden? } if options[:exclude_hidden_objects]
```

Comment:
`ResourcePool` has a helper method `:hidden?` which checks `is_default?` so I only had to reject one thing here which made it much cleaner
https://github.com/ManageIQ/manageiq/blob/master/app/models/resource_pool.rb#L36